homeassistant/components/harmony/config_flow.py

```diff
@@ -0,0 +1,187 @@
+"""Config flow for Logitech Harmony Hub integration."""
+import logging
+from urllib.parse import urlparse
+
+import aioharmony.exceptions as harmony_exceptions
+from aioharmony.harmonyapi import HarmonyAPI
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.components import ssdp
+from homeassistant.components.remote import (
+    ATTR_ACTIVITY,
+    ATTR_DELAY_SECS,
+    DEFAULT_DELAY_SECS,
+)
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import callback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_HOST): str, vol.Required(CONF_NAME): str}, extra=vol.ALLOW_EXTRA
```

Comment:
Name is a construct meant for configuration.yaml and shouldn't be a part of a config entry, use harmony.name instead

Reply:
I used harmony.name as a last resort because its usually the ip address of the device or worse the string `""`

Reply:
Added a comment to explain this in the code for future devs